pkgs/coverage/lib/src/coverage_options.dart

```diff
@@ -54,24 +69,48 @@ class CoverageOptionsProvider {
       fileSourceUri: file?.uri,
     );
 
-    coverageOptions = CoverageOptions.fromConfig(options, defaultOptions);
+    coverageOptions =
+        CoverageOptions.fromConfig(options, defaultOptions, optionsFilePath);
   }
 
   late final CoverageOptions coverageOptions;
+  late final String optionsFilePath;
+  static const defaultFilePath = 'coverage_options.yaml';
+
+  File? _getOptionsFile(String? filePath) {
+    filePath ??= _findOptionsFilePath();
+    if (filePath == null) {
+      throw StateError('Could not find a pubspec.yaml file.');
```

Comment:
Just return null in this case. Don't want to break existing users who are running their commands from outside a package directory.